### PR TITLE
fix(editor): Trial banner does not disappear after sign out (no-changelog)

### PR DESCRIPTION
**Pull request by RicardoE105**
to test in staging use version `PR-6930-ado-990-trial-banner-does-not-disappear-after-sign-out`

image

### CODE REVIEW

**Review by MiloradFilipovic — DISMISSED** [review]
Nice job! Found just one typo

File: packages/editor-ui/src/stores/ui.store.ts [hunk 1]
```diff
@@ -565,5 +565,23 @@ export const useUIStore = defineStore(STORES.UI, {
...
+			if (cloudPlanStore.userIsTrialing) {
+				await this.dismissBanner('V1', 'temporary');
+				if (cloudPlanStore.trialExpired) {
+					this.showBanner('TRIAL_OVER');
+				} else {
+					this.showBanner('TRIAL');
+				}
+			}
+		},
+		async dismissAllBanner() {
```
[the remark this answers is not in the document]
Comment (RicardoE105) [reply packages/editor-ui/src/stores/ui.store.ts]: thanks just fixed that. Will add the e2e then!

File: packages/editor-ui/src/App.vue [hunk 2]
```diff
@@ -262,9 +239,7 @@ export default defineComponent({
 		await this.redirectIfNecessary();
 		void this.checkForNewVersions();
 		await this.checkForCloudPlanData();
-		await this.initBanners();
-
-		void this.checkForCloudPlanData();
+		void this.initBanners();
```
Comment (mutdmour) [inline packages/editor-ui/src/App.vue:242]: as discussed in grooming, would be great to add tests that the upgrade bar shows up

Comment (RicardoE105) [reply packages/editor-ui/src/App.vue:242]: yes! Will do. Wanted to make sure the approach was correct before adding the tests!

Comment (mutdmour) [reply packages/editor-ui/src/App.vue:242]: thanks

File: cypress/e2e/27-opt-in-trial-banner.cy.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+const mainSidebar = new MainSidebar();
+const bannerStack = new BannerStack();
+const signinPage = new SigninPage();
+const workflowPage = new WorkflowPage();
+
+describe('BannerStack', { disableAutoLogin: true }, () => {
+	before(() => {
+		const now = new Date();
+		const fiveDaysFromNow = new Date(now.getTime() + 5 * 24 * 60 * 60 * 1000);
```
Comment (MiloradFilipovic) [inline cypress/e2e/27-opt-in-trial-banner.cy.ts:13]: Not very important, but could be a bit cleaner if we use luxon `DateTime` and [`plus` function](https://moment.github.io/luxon/api-docs/index.html#datetimeplus)

**Review by MiloradFilipovic — APPROVED** [review]
Tests look good. Added one minor comment but good to go without it

**Review by MiloradFilipovic — CHANGES_REQUESTED** [review]
Dismissing  approval until we fix the syntax error reported [above](https://github.com/n8n-io/n8n/pull/6930#issuecomment-1681873464)

**Comment by RicardoE105** [comment]
@MiloradFilipovic 

> I guess the disableAutoLogin parameter not supported in this place.

We are using the autologin in other two places in the E2E tests


image

image

File: cypress/e2e/27-opt-in-trial-banner.cy.ts (hunk 3 above)
Comment (RicardoE105) [reply cypress/e2e/27-opt-in-trial-banner.cy.ts:12]: Did not want to use a dependency only to add 5 days to a date!

**Review by MiloradFilipovic — APPROVED** [review]
All good now. Thanks for addressing

**Comment by janober** [comment]
Got released with [`n8n@1.3.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.3.1)

**Pull request merged**